_source/_posts/2022-06-06-secure-databases-dotnet.md

```diff
@@ -0,0 +1,650 @@
+
+Next, replace the code in `Program.cs` with the following.
+
+```csharp
+using Microsoft.AspNetCore.Authentication.Cookies;
+using Microsoft.AspNetCore.Authentication.OpenIdConnect;
+using Okta.AspNetCore;
+
+var builder = WebApplication.CreateBuilder(args);
+
+// Add services to the container.
+builder.Services.AddControllersWithViews();
+
+builder.Services.ConfigureApplicationCookie(options =>
+{
+    options.Cookie.HttpOnly = true;
+    options.Cookie.SecurePolicy = Microsoft.AspNetCore.Http.CookieSecurePolicy.Always;
+})
+.AddAuthentication(options =>
+{
+    options.DefaultScheme = CookieAuthenticationDefaults.AuthenticationScheme;
+    options.DefaultChallengeScheme = OpenIdConnectDefaults.AuthenticationScheme;
+})
+.AddCookie()
+.AddOktaMvc(new OktaMvcOptions
+{
+    // Replace these values with your Okta configuration
+    OktaDomain = builder.Configuration.GetValue<string>("Okta:Domain"),
+    ClientId = builder.Configuration.GetValue<string>("Okta:ClientId"),
+    ClientSecret = builder.Configuration.GetValue<string>("Okta:ClientSecret"),
+    Scope = new List<string> { "openid", "profile", "email" },
+    PostLogoutRedirectUri = "/"
+});
+
+builder.Services.AddScoped<Okta_CosmosDb.Services.IScrubService, Okta_CosmosDb.Services.ScrubService>();
+builder.Services.AddSingleton<Okta_CosmosDb.Services.ICosmosService>(Okta_CosmosDb.Services.CosmosService.InitializeCosmosClientInstanceAsync(builder.Configuration.GetSection("CosmosDb")).GetAwaiter().GetResult());
+
+var app = builder.Build();
+
+// Configure the HTTP request pipeline.
+if (!app.Environment.IsDevelopment())
+{
+    app.UseExceptionHandler("/Home/Error");
+    // The default HSTS value is 30 days. You may want to change this for production scenarios, see https://aka.ms/aspnetcore-hsts.
+    app.UseHsts();
+}
+
+app.UseHttpsRedirection();
+app.UseStaticFiles();
+
+app.UseRouting();
+
+app.UseAuthentication();
+app.UseAuthorization();
+
+app.MapControllerRoute(
+    name: "default",
+    pattern: "{controller=Home}/{action=Index}/{id?}");
+
+app.Run();
+```
+
+Much of this code is boilerplate from the Visual Studio scaffolding process but you are adding a few extra items.
+
+First, you need to configure your authentication to use `AddOktaMvc` and provide the configuration values from your `appsettings` file.
+
+Next, you are setting up the dependency injection for a couple of custom services that you will write soon. These are the `IScrubService` which will handle your scrubbing process and the `ICosmosService` which will handle communication to your Cosmos DB account.
+
+Finally, you are calling `InitializeCosmosClientInstanceAsync` which will set up your database and container if they do not exist. Then it will return the `CosmosService` as a singleton.  
```

Comment:
```suggestion
Finally, you are calling `InitializeCosmosClientInstanceAsync`, which will set up your database and container if they do not exist. Then it will return the `CosmosService` as a singleton.  
```